packages/block-editor/src/components/link-control/index.js

```diff
@@ -77,9 +78,22 @@ function LinkControl( {
 	};
 
 	// Utils
-	const startEditMode = () => {
-		if ( isFunction( onLinkChange ) ) {
-			onLinkChange();
+
+	/**
+	 * Handler function which turns edit mode ON.
+	 * Also, it calls `onEdit` prop.
+	 */
+	const onEditHandler = () => {
+		setIsEditingLink( true );
+
+		// Populate input searcher whether
+		// the current link has a title.
+		if ( currentLink && currentLink.title ) {
+			setInputValue( currentLink.title );
+		}
+
+		if ( isFunction( onEdit ) ) {
+			onEdit();
```

Comment:
So because we are no longer expecting `currentLink` to be changed when someone hits "Change" we need a way to signal that we're now back editing the link.